### DISCUSSION TITLE
How would you use modifiers like .single()?

### DISCUSSION THREAD

**Discussion by cdedreuille** [tmm/react-supabase#25] [Q&A]
Hi,

I querying a single object and I found it quite handy to use `.single()` for that.
I tried to add it to the filter like this but Typescript is not happy about it. I get the result correctly though.

```
const [{ data: product }] = useSelect('products', {
    filter: useFilter((q) => q.eq('id', query.productId).single(), [query]),
    pause: !isReady
  });
```

What would be the best way to tackle this? Should this be part of the options in `useSelect()`?

**Comment by bruceharrison1984** [comment]
I'm curious to this as well. Even more because [this question](https://github.com/tmm/react-supabase/discussions/7#discussioncomment-765988) by the author shows `single` being used in the filter.
